File: ModFolder/Loenn/entities/ReskinnableBadelineBoost.lua
local badelineBoost = {}

badelineBoost.name = "LuckyHelper/ReskinnableBadelineBoost"
badelineBoost.depth = -1000000
badelineBoost.nodeLineRenderType = "line"
badelineBoost.texture = "objects/badelineboost/idle00"
badelineBoost.nodeLimits = {0, -1}
badelineBoost.placements = {
    name = "normal",
    data = {
        lockCamera = true,
        canSkip = false,
        finalCh9Boost = false,
        finalCh9GoldenBoost = false,
        finalCh9Dialog = false,
        XMLId = "badelineBoost",
        stretchTexturePath = "objects/badelineboost/stretch"
    }
}

return badelineBoost

File: Entities/Misc/ReskinnableBadelineBoost.cs
using Celeste.Mod.Entities;
using LuckyHelper.Extensions;
using LuckyHelper.Module;
using Microsoft.Xna.Framework.Graphics;

namespace LuckyHelper.Entities;

[CustomEntity("LuckyHelper/ReskinnableBadelineBoost")]
[Tracked]
public class ReskinnableBadelineBoost : BadelineBoost
{
    public ReskinnableBadelineBoost(Vector2[] nodes, bool lockCamera, bool canSkip = false, bool finalCh9Boost = false, bool finalCh9GoldenBoost = false, bool finalCh9Dialog = false) : base(nodes, lockCamera, canSkip, finalCh9Boost, finalCh9GoldenBoost, finalCh9Dialog)
    {
    }

    public ReskinnableBadelineBoost(EntityData data, Vector2 offset) : base(data, offset)
    {
        Remove(sprite);
        sprite = GFX.SpriteBank.Create(data.Attr("XMLId"));
        Add(sprite);
        stretch.Texture = GFX.Game[data.Attr("stretchTexturePath")];
    }
}